### ISSUE TITLE
Is an e1 detector occupied only when a vehicle occupies the center of the detector?

### ISSUE THREAD

**Issue by MohamadReza-Haghani** [eclipse/sumo#10820]
Hi,

I am using an e1 detector. I reduced the simulation time step length to 0.1 sec to approximate collecting event-based data using a detector. I have two questions. Firstly, I was wondering if there is a better way to measure event-based data. Secondly, when I use the `getLastStepOccupancy` command in Traci to get the occupancy of the detector, I noticed that it only considers the detector occupied if only a vehicle occupies the center of the detector. I was wondering how I can change the setting of the detector to consider the detector occupied if only a small portion of the detector is occupied. In the attached screenshot, I got 0 as the result of `getLastStepOccupancy.`

Best,
Reza.
![Capture2](https://user-images.githubusercontent.com/64617116/170901229-38fb6dfc-9f3f-482b-a34d-146b45674ecf.JPG)

**Comment by m-kro** [comment]
Since SUMO v.1.13.0, E1 detectors may have a length > 0, counting downstream from the detector reference position (see [docs](https://sumo.dlr.de/docs/Simulation/Output/Induction_Loops_Detectors_%28E1%29.html)). By default, their length is still 0, so that it counts only if a vehicle occupies the road at the reference position.
Detectors are quite the standard way to measure local vehicle data in SUMO and to process it further (whatever your event may be).

**Comment by MohamadReza-Haghani** [comment]
Thank you for the response! So, I should use an e2 detector if I want the detector to detect vehicles in an area.

**Comment by m-kro** [comment]
Yes if you use an older SUMO version than v.1.13.0, otherwise you could use the E1 detector and specify the length by setting the attribute of the same name.

**Comment by namdre** [comment]
> Since SUMO v.1.13.0, E1 detectors may have a length > 0, counting downstream from the detector reference position (see [docs](https://sumo.dlr.de/docs/Simulation/Output/Induction_Loops_Detectors_%28E1%29.html)). 

This feature is only available in the development version but not yet in 1.13.0
